math/libfaiss/Portfile

```diff
@@ -24,39 +22,134 @@ long_description    Library for efficient similarity search and clustering of \
+                        port:swig-python
+    depends_lib-append  port:${name} \
+                        port:py${python.version}-numpy
+
+    post-extract {
+        # fix destroot commands like `$(PYTHON) setup.py install`
+        reinplace -E \
+            "s|^(\[\[:space:]]+\\\$\\(PYTHON\\)\[\[:space:]]+setup\.py\[\[:space:]]+)(install)|\\1 --no-user-cfg \\2 ${destroot.destdir}|" \
+            ${worksrcpath}/python/Makefile
+    }
+
+    use_configure       yes
+
+    configure.env-append \
+                        PYTHON=${python.bin}
+
+    configure.args-append \
+                        --with-python=${python.bin} \
+                        --with-swig=${prefix}/bin/swig
+
+    pre-build {
+        system -W ${worksrcpath} ${build.cmd}
+    }
+
+    build.cmd           [portbuild::build_getmaketype]
+    build.target        all
+    build.pre_args      -C python ${build.target}
+
+    destroot.cmd        ${build.cmd}
+    destroot.target     install
+    destroot.pre_args   -C python ${destroot.target}
+    destroot.post_args
+
+    post-destroot {
+        set docdir ${prefix}/share/doc/${subport}
+        set tutorialdir ${prefix}/share/${subport}/tutorial
+        xinstall -d \
+            ${destroot}${docdir} \
+            ${destroot}${tutorialdir}
+        xinstall -m 0644 -W ${worksrcpath} README.md LICENSE \
+            ${destroot}${docdir}
+        copy ${worksrcpath}/tutorial/python \
+            ${destroot}${tutorialdir}
+    }
+
+    test.run            yes
+    test.dir            ${build.dir}/tests
+    test.cmd            ${python.bin}
+    test.pre_args       "-c 'import subprocess as sp; \
+                            printout = lambda scpt: print(scpt+\":\\n\"+sp.run(\[\"${python.bin}\", scpt],stdout=sp.PIPE,stderr=sp.STDOUT).stdout.decode()); \
+                            test_scripts = \[ \
+                                \"test_binary_factory.py\", \
+                                \"test_binary_io.py\", \
+                                \"test_extra_distances.py\", \
+                                \"test_factory.py\", \
+                                \"test_index.py\", \
+                                \"test_index_accuracy.py\", \
+                                \"test_index_binary.py\", \
+                                \"test_index_binary_from_float.py\", \
+                                \"test_ivflib.py\", \
+                                \"test_meta_index.py\", \
+                                \"test_omp_threads_py.py\", \
+                                \"test_oom_exception.py\", \
+                                \"test_referenced_objects.py\", \
+                                \"test_standalone_codec.py\"]; \
+                            \[ printout(scpt) for scpt in test_scripts ];'"
+    test.target
+    test.env-append \
+                    "PATH=$env(PATH):${python.prefix}/bin" \
+                    PYTHONPATH=${worksrcpath}/build/lib
```

Comment:
the tests fail with `ModuleNotFoundError: No module named 'faiss'` - in the PR checklist it states that the tests work (it might if you have the port already installed though). It should likely be `PYTHONPATH=${worksrcpath}/build/lib` instead.

Reply:
Fixed thanks.